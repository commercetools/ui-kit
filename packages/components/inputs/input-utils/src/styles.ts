import { css } from '@emotion/react';
import { designTokens } from '@commercetools-uikit/design-system';

type TInputProps = {
  isDisabled?: boolean;
  disabled?: boolean;
  hasError?: boolean;
  hasWarning?: boolean;
  isReadOnly?: boolean;
  readOnly?: boolean;
};

const getInputBorderColor = (
  props: TInputProps,
  defaultBorderColor: string = designTokens.borderColorForInput
) => {
  if (props.isDisabled || props.disabled) {
    return designTokens.borderColorForInputWhenDisabled;
  }
  if (props.hasError) {
    return designTokens.borderColorForInputWhenError;
  }
  if (props.hasWarning) {
    return designTokens.borderColorForInputWhenWarning;
  }
  if (props.isReadOnly || props.readOnly) {
    return designTokens.borderColorForInputWhenReadonly;
  }
  return defaultBorderColor;
};

const getInputFontColor = (props: TInputProps) => {
  if (props.isDisabled || props.disabled) {
    return designTokens.fontColorForInputWhenDisabled;
  }
  if (props.hasError) {
    return designTokens.fontColorForInputWhenError;
  }
  if (props.hasWarning) {
    return designTokens.fontColorForInputWhenWarning;
  }
  if (props.isReadOnly || props.readOnly) {
    return designTokens.fontColorForInputWhenReadonly;
  }
  return designTokens.fontColorForInput;
};

const getInputBorderWidth = (props: TInputProps) => {
  if (props.hasError) {
    return designTokens.borderWidthForInputWhenError;
  }
  if (props.hasWarning) {
    return designTokens.borderWidthForInputWhenWarning;
  }
  return designTokens.borderWidthForInput;
};

const getInputBoxShadow = (props: TInputProps) => {
  if (props.hasError) {
    return designTokens.shadowForInputWhenError;
  }
  if (props.hasWarning) {
    return designTokens.shadowForInputWhenWarning;
  }
  return designTokens.shadowForInput;
};

const getInputStyles = (props: TInputProps) => {
  return css`
    appearance: none;
    background-color: ${props.isDisabled || props.disabled
      ? designTokens.backgroundColorForInputWhenDisabled
      : designTokens.backgroundColorForInput};
    border: ${getInputBorderWidth(props)} solid ${getInputBorderColor(props)};
    border-radius: ${designTokens.borderRadiusForInput};
    box-sizing: border-box;
    box-shadow: ${getInputBoxShadow(props)};
    color: ${getInputFontColor(props)};
    cursor: ${props.isDisabled ? 'not-allowed' : 'default'};
    display: flex;
    flex: 1;
    font-family: inherit;
    font-size: ${designTokens.fontSizeForInput};
    height: ${designTokens.heightForInput};
    min-height: ${designTokens.heightForInput};
    opacity: ${props.isDisabled || props.disabled
      ? '1'
      : 'unset'}; /* fix for mobile safari */
    outline: none;
    overflow: hidden;
    padding: 0 ${designTokens.paddingForInput};
    transition: border-color ${designTokens.transitionStandard},
      background-color ${designTokens.transitionStandard},
      color ${designTokens.transitionStandard},
      box-shadow ${designTokens.transitionStandard};
    width: 100%;

    &::placeholder {
      color: ${designTokens.placeholderFontColorForInput};
    }
    :active,
    :focus,
    :hover:not(:disabled):not(:read-only):not(:focus) {
      border-color: ${getInputBorderColor(
        props,
        designTokens.borderColorForInputWhenHovered
      )};
      background-color: ${designTokens.backgroundColorForInputWhenHovered};
    }
    :focus {
      box-shadow: ${designTokens.shadowForInputWhenFocused};
      border-color: ${designTokens.borderColorForInputWhenFocused};
      background-color: ${designTokens.backgroundColorForInputWhenFocused};
    }
  `;
};

// eslint-disable-next-line import/prefer-default-export
export { getInputStyles };
