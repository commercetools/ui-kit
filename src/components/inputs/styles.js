import { css } from '@emotion/core';
import vars from '../../../materials/custom-properties';
import designTokens from '../../../materials/design-tokens';

const getInputStyles = (props, theme) => {
  const overwrittenVars = {
    ...vars,
    ...theme,
  };

  const baseStyles = css`
    appearance: none;
    background-color: ${overwrittenVars[designTokens.backgroundColorForInput]};
    border: 1px solid ${overwrittenVars[designTokens.borderColorForInput]};
    border-radius: ${overwrittenVars[designTokens.borderRadiusForInput]};
    box-shadow: none;
    box-sizing: border-box;
    color: ${overwrittenVars[designTokens.fontColorForInput]};
    display: flex;
    flex: 1;
    font-family: ${overwrittenVars.fontFamilyDefault};
    font-size: ${overwrittenVars[designTokens.fontSizeForInput]};
    height: ${overwrittenVars.sizeHeightInput};
    min-height: ${overwrittenVars.sizeHeightInput};
    outline: none;
    overflow: hidden;
    padding: 0 ${overwrittenVars.spacingS};
    transition: border-color ${overwrittenVars.transitionStandard},
      background-color ${overwrittenVars.transitionStandard},
      color ${overwrittenVars.transitionStandard};
    width: 100%;

    &::placeholder {
      color: ${overwrittenVars[designTokens.placeholderFontColorForInput]};
    }
    &:active,
    &:focus {
      border-color: ${overwrittenVars[
        designTokens.borderColorForInputWhenFocused
      ]};
    }
  `;
  if (props.disabled || props.isDisabled) {
    return [
      baseStyles,
      css`
        background-color: ${overwrittenVars[
          designTokens.backgroundColorForInputWhenDisabled
        ]};
        border-color: ${overwrittenVars[
          designTokens.borderColorForInputWhenDisabled
        ]};
        color: ${overwrittenVars[designTokens.fontColorForInputWhenDisabled]};
        cursor: not-allowed;
        opacity: 1; /* fix for mobile safari */
      `,
    ];
  }
  if (props.hasError) {
    return [
      baseStyles,
      css`
        border-color: ${overwrittenVars[
          designTokens.borderColorForInputWhenError
        ]};
        color: ${overwrittenVars[designTokens.fontColorForInputWhenError]};
      `,
    ];
  }
  if (props.hasWarning) {
    return [
      baseStyles,
      css`
        border-color: ${overwrittenVars[
          designTokens.borderColorForInputWhenWarning
        ]};
        color: ${overwrittenVars[designTokens.fontColorForInputWhenWarning]};
      `,
    ];
  }
  if (props.readOnly || props.isReadOnly) {
    return [
      baseStyles,
      css`
        border-color: ${overwrittenVars[
          designTokens.borderColorForInputWhenReadonly
        ]};
        color: ${overwrittenVars[designTokens.fontColorForInputWhenReadonly]};
        cursor: default;
      `,
    ];
  }
  return baseStyles;
};

// eslint-disable-next-line import/prefer-default-export
export { getInputStyles };
