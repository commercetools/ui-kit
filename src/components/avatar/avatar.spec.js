import React from 'react';
import { shallow } from 'enzyme';
import Avatar, { Initials, GravatarImg } from './avatar';
import styles from './avatar.mod.css';

const createTestProps = customProps => ({
  firstName: '',
  lastName: '',
  gravatarHash: '20c9c1b252b46ab49d6f7a4cee9c3e68',
  isHighlighted: false,
  scale: 'l',
  ...customProps,
});

describe('render', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<Avatar {...props} />);
  });

  it('should have base style', () => {
    expect(wrapper).toHaveClassName(styles['avatar-base']);
  });

  it('should render <GravatarImg>', () => {
    expect(wrapper).toRender(GravatarImg);
  });

  it('should render <Initials>', () => {
    expect(wrapper).toRender(Initials);
  });

  describe('<GravatarImg />', () => {
    describe('with regular email', () => {
      beforeEach(() => {
        wrapper = shallow(
          <GravatarImg scale="s" hash="20c9c1b252b46ab49d6f7a4cee9c3e68" />
        );
      });

      it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('when large scale', () => {
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<Avatar {...props} />);
    });

    it('should contain style for large scale', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when medium scale', () => {
    beforeEach(() => {
      props = createTestProps({ scale: 'm' });
      wrapper = shallow(<Avatar {...props} />);
    });
    it('should contain style for medium scale', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when small scale', () => {
    beforeEach(() => {
      props = createTestProps({ scale: 's' });
      wrapper = shallow(<Avatar {...props} />);
    });
    it('should contain style for small scale', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when highlighted', () => {
    beforeEach(() => {
      props = createTestProps({ isHighlighted: true });
      wrapper = shallow(<Avatar {...props} />);
    });
    it('should contain style for highligting component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
