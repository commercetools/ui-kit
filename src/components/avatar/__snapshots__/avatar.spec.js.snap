// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <GravatarImg /> with regular email should match snapshot 1`] = `
<img
  className="gravatar-img gravatar-image-small"
  src="https://www.gravatar.com/avatar/20c9c1b252b46ab49d6f7a4cee9c3e68?d=blank"
  srcSet="https://www.gravatar.com/avatar/20c9c1b252b46ab49d6f7a4cee9c3e68?d=blank"
/>
`;

exports[`render when highlighted should contain style for highligting component 1`] = `
<div
  className="avatar-base avatar-l avatar-hover"
>
  <GravatarImg
    hash="20c9c1b252b46ab49d6f7a4cee9c3e68"
    scale="l"
  />
  <Initials
    firstName=""
    lastName=""
  />
</div>
`;

exports[`render when large scale should contain style for large scale 1`] = `
<div
  className="avatar-base avatar-l"
>
  <GravatarImg
    hash="20c9c1b252b46ab49d6f7a4cee9c3e68"
    scale="l"
  />
  <Initials
    firstName=""
    lastName=""
  />
</div>
`;

exports[`render when medium scale should contain style for medium scale 1`] = `
<div
  className="avatar-base avatar-m"
>
  <GravatarImg
    hash="20c9c1b252b46ab49d6f7a4cee9c3e68"
    scale="m"
  />
  <Initials
    firstName=""
    lastName=""
  />
</div>
`;

exports[`render when small scale should contain style for small scale 1`] = `
<div
  className="avatar-base avatar-s"
>
  <GravatarImg
    hash="20c9c1b252b46ab49d6f7a4cee9c3e68"
    scale="s"
  />
  <Initials
    firstName=""
    lastName=""
  />
</div>
`;
